packages/dapp/components/lib/hooks/useWeb3.tsx

```diff
@@ -50,13 +60,54 @@ export const UseWeb3Provider: React.FC<ChildrenShim> = ({ children }) => {
   const [web3State, setWeb3State] = useState<Web3State>(DEFAULT_WEB3_STATE);
 
   useEffect(() => {
-    if (storedProviderMode === "jsonrpc" && storedWallet) {
-      connectJsonRpc(storedWallet);
-    } else {
-      connectMetaMask();
+    if (storedProviderMode) {
+      if ("jsonrpc" == storedProviderMode) {
+        if (storedWallet) {
+          connectJsonRpc(storedWallet);
+        }
+      } else if ("metamask" == storedProviderMode) {
+        connectMetaMask();
+      }
     }
   }, []);
 
+  async function connectWalletConnect() {
+    const bridge = "https://bridge.walletconnect.org";
+
+    // create new connector
+    const connector = new WalletConnect({ bridge, qrcodeModal: QRCodeModal });
+
+    // check if already connected
+    if (!connector.connected) {
+      // create new session
+      await connector.createSession();
+    }
+
+    connector.on("connect", async (error, payload) => {
+      const { chainId, accounts } = payload.params[0];
+      const newWalletAddress = accounts[0];
+
+      const newProvider = new WalletConnectProvider({
+        infuraId: process.env.API_KEY_ALCHEMY,
```

Comment:
This might be wrong. Alchemy is a different service with infura.io. WalletConnectProvider supports 2 available options (rpc & infura). iirc, they don't support alchemy right now.

Reply:
Okay, will recreate infura ID and rewrite down